### PR TITLE
Comment on shared fd's (out/in) is added

### PR DESCRIPTION
**Pull request by gc00**
I finally figured out how to set up the discussion on the comments as a pull request --- _even though I had accidentally pushed the comments into origin at an earlier time_.  I'll respond to the existing comments by @karya0 soon.

### CODE REVIEW

**Comment by gc00** [comment]
@karya0:  I've updated the comments.  Is this okay to commit?

File: src/plugin/ipc/connectionlist.cpp [hunk 1]
```diff
@@ -38,17 +38,44 @@
...
+//  state of a file descriptor (whether shared or not) at the time of restart.
+//  It is implemented through doLocking and checkLocking, in which each
+//  process uses fcntl() with SETOWN, and after a barrier, they check
+//  if they are still the owner using GETOWN.
+// A list of connections (fd's) is found through /proc/*/fd and saved
+//  in the ConnectionList object.
+// At the time of restart, con->hasLock() will tell a process if it is the
+//  the leader for that connection (con).  However, the process must still
+//  discover if the file descriptor is private or shared.
+// We define a "private fd" as an fd that is shared by exactly one process.
```
Comment (karya0) [inline src/plugin/ipc/connectionlist.cpp]: Nit: Can you move this into the first para?

**Comment by karya0** [comment]
It's ready to be committed, however, the target branch is dmtcp/dmtcp:comments_dont_exist instead of dmtcp/dmtcp:master, so let's make sure that you commit to dmtcp/dmtcp:master and kill the other branch.

**Comment by gc00** [comment]
I successfully committed to dmtcp/dmtcp:master even though my local master was two commits ahead!  I think I'm starting to get the hang of this.  :-)  Thanks.

**Pull request closed without merging**